libs/src/service-selection/ServiceSelection.ts

```diff
@@ -162,33 +222,28 @@ class ServiceSelection {
    * @param {boolean} verbose whether or not to return full services metadata
    * @param {Set} whitelist a whitelist to override the set of endpoints
    */
-  async findAll ({ verbose = false, whitelist = null } = {}) {
+  async findAll ({ verbose = false, whitelist = this.whitelist } = {}) {
     // Get all the services
-    let services = await this.getServices({ verbose })
+    const services = await this.getServices({ verbose })
+
+    const baseServices = services.map(service => isVerbose(service) ? service.endpoint : service)
 
     // If a whitelist is provided, filter down to it
-    whitelist = whitelist || this.whitelist
     if (whitelist) {
-      // If using verbose, the services are an object with service metadata, not a string endpoint
-      if (verbose) {
-        services = services.filter(s => whitelist.has(s.endpoint))
-      } else {
-        services = services.filter(s => whitelist.has(s))
-      }
+      baseServices.filter(service => whitelist.has(service))
     }
 
     // Key the services by their health check endpoint
-    const map = services.reduce((acc, s) => {
-      const serviceEndpoint = verbose ? s.endpoint : s
-      acc[ServiceSelection.getHealthCheckEndpoint(serviceEndpoint)] = s
+    const urlMap = baseServices.reduce<Record<string, string>>((acc, s) => {
```

Comment:
```suggestion
    const urlMap = baseServices.reduce<Record<string, string>>((map, serviceEndpoint) => {
```

Reply:
yeah good call, maybe we just go with urlMap even?